### PR TITLE
Elasticsearch hhtps

### PR DESCRIPTION
**Pull request by maggarwal13**

### CODE REVIEW

File: whois-api/src/main/java/net/ripe/db/whois/api/elasticsearch/ElasticSearchInstance.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+@Profile({WhoisProfile.DEPLOYED})
+@Primary
+@Component
+public class ElasticSearchInstance implements ElasticRestHighlevelClient {
+    private final RestHighLevelClient client;
+
```
Comment (MiguelAHM) [inline whois-api/src/main/java/net/ripe/db/whois/api/elasticsearch/ElasticSearchInstance.java:27]: Should we add the logger here as well?
private static final Logger LOGGER = getLogger(ElasticIndexService.class);

Comment (maggarwal13) [reply whois-api/src/main/java/net/ripe/db/whois/api/elasticsearch/ElasticSearchInstance.java:27]: Do we need to log something here?

File: whois-api/src/test/java/net/ripe/db/whois/api/elasticsearch/ElasticSearchTestInstance.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import javax.annotation.Nullable;
+import java.util.List;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+@Profile({WhoisProfile.TEST})
+@Component
+public class ElasticSearchTestInstance implements ElasticRestHighlevelClient {
+
+    private static final Logger LOGGER = getLogger(ElasticIndexService.class);
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/elasticsearch/ElasticSearchTestInstance.java:24]: Change LOGGER class to `ElasticSearchTestInstance`

File: whois-api/src/test/java/net/ripe/db/whois/api/elasticsearch/ElasticSearchTestInstance.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    private final RestHighLevelClient client;
+
+    @Autowired
+    public ElasticSearchTestInstance(@Value("#{'${elastic.host:}'.split(',')}") final List<String> elasticHosts) {
+        this.client = getEsClient(elasticHosts);
+    }
+
+    @Nullable
+    private RestHighLevelClient getEsClient(final List<String> elasticHosts) {
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/elasticsearch/ElasticSearchTestInstance.java:33]: Can you name the class according to what it does? It looks like this is the "non auth" instance vs. `ElasticSearchInstance` which requires authentication.

File: whois-api/src/main/java/net/ripe/db/whois/api/elasticsearch/ElasticSearchInstance.java (hunk 1 above)
Comment (maggarwal13) [reply whois-api/src/main/java/net/ripe/db/whois/api/elasticsearch/ElasticSearchInstance.java:27]: Actually I should catch all exception here and log as warning.

File: whois-api/src/test/java/net/ripe/db/whois/api/elasticsearch/ElasticSearchTestInstance.java (hunk 3 above)
Comment (maggarwal13) [reply whois-api/src/test/java/net/ripe/db/whois/api/elasticsearch/ElasticSearchTestInstance.java:33]: May be ElasticSearchTestNonAuthInstance? It is used only in test and it is in test package so I named it as ElasticSearchTestInstance

Comment (maggarwal13) [reply whois-api/src/test/java/net/ripe/db/whois/api/elasticsearch/ElasticSearchTestInstance.java:33]: Renamed to ElasticSearchNonAuthInstance

Comment (eshryane) [reply whois-api/src/test/java/net/ripe/db/whois/api/elasticsearch/ElasticSearchTestInstance.java:33]: Thanks @maggarwal13 I think this name is clearer as it describes what it does (but still stays in `src/test/`)

**Review by eshryane — APPROVED** [review]

**Pull request closed without merging**